lib/api/core/validation/index.js

```diff
@@ -187,17 +209,37 @@ class Validation {
         specSubset = collectionSpecSubset[fieldName],
         field = documentSubset[fieldName];
 
-      if (documentSubset.hasOwnProperty(fieldName) && this.types[specSubset.type].allowChildren && specSubset.children) {
+      if (
+        _.has(documentSubset, fieldName)
+        && this.types[specSubset.type].allowChildren && specSubset.children
+      ) {
         if (Array.isArray(field)) {
           for (let i = 0; i < field.length; i++) {
-            field[i] = this.recurseApplyDefault(isUpdate, field[i], specSubset.children);
+            field[i] = this.recurseApplyDefault(
+              isUpdate,
+              field[i],
+              specSubset.children);
           }
+        } else {
+          documentSubset[fieldName] = this.recurseApplyDefault(
+            isUpdate,
+            field,
+            specSubset.children
+          );
         }
         else {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Parse error at line 230 column 9:

220:               field[i],
221:               specSubset.children);
222:           }
223:         } else {
224:           documentSubset[fieldName] = this.recurseApplyDefault(
225:             isUpdate,
226:             field,
227:             specSubset.children
228:           );
229:         }
230:         else {
             ^
231:           documentSubset[fieldName] = this.recurseApplyDefault(
232:             isUpdate,
233:             field,
234:             specSubset.children);
235:         }
236:       }
237:       else if (
238:         specSubset.defaultValue
239:         && (
240:           field === null [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AParsingError)